frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts

```diff
@@ -0,0 +1,88 @@
+import { DatePipe } from '@angular/common';
+import { Component } from '@angular/core';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
+import { MatCalendar, MatDatepicker } from '@angular/material/datepicker';
+import * as moment from 'moment';
+
+@Component({
+    selector: 'app-suitable-lesson',
+    templateUrl: './suitable-lesson.component.html',
+    styleUrls: ['./suitable-lesson.component.sass'],
+})
+export class SuitableLessonComponent {
+    days: Date[] = [];
+
+    selectedDate: Date = new Date();
+
+    isSelected = (date: Date): string | string[] => {
+        if (this.selectedDate && date.getDate() === this.selectedDate.getDate()) {
+            return 'selected';
+        }
+
+        return '';
+    };
+
+    constructor() {
+        this.setDays();
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
+    select(event: Date | null, calendar: MatCalendar<Date>) {
+        if (event !== null) {
+            this.selectedDate = event;
+            this.selectedDate.setDate(this.selectedDate.getDate() - 1);
+            this.setDays();
+        }
+    }
+
+    weekInc(): void {
+        this.selectedDate.setDate(this.selectedDate.getDate() + 7);
+
+        this.setDays();
+    }
+
+    weekDec(): void {
+        this.selectedDate.setDate(this.selectedDate.getDate() - 7);
+
+        this.setDays();
+    }
+
+    getWeekDayString(inDate: Date): string {
+        return new DatePipe('en-US').transform(inDate, 'EEEE') ?? '';
```

Comment:
Please do not use magic numbers

Reply:
Still not fixed
Move amount of days in week (1/7) to const variables at the top of the class